src/views/Posters/index.jsx

```diff
@@ -0,0 +1,553 @@
+
+  useEffect(() => {
+    const loadPosters = async () => {
+      const posters = await getCurrentPosters();
+      setAllPosters(posters);
+    };
+    loadPosters();
+  }, []);
+
+  useEffect(() => {
+    const loadButton = async () => {
+      if (sessionFromURL) {
+        setSelectedSession(sessionService.encodeSessionCode(sessionFromURL));
+      } else {
+        const { SessionCode: currentSessionCode } = await sessionService.getCurrent();
+        const [involvements, sessions] = await Promise.all([
+          involvementService.getAll(currentSessionCode),
+          sessionService.getAll(),
+        ]);
+        if (involvements.length === 0) {
+          let IndexOfCurrentSession = sessions.findIndex(
+            (session) => session.SessionCode === currentSessionCode,
+          );
+          for (let k = IndexOfCurrentSession + 1; k < sessions.length; k++) {
+            const newInvolvements = await involvementService.getAll(sessions[k].SessionCode);
+            if (newInvolvements.length !== 0) {
+              setSelectedSession(sessions[k].SessionCode);
+              break;
+            }
+          }
+        } else {
+          setSelectedSession(currentSessionCode);
+        }
+      }
+    };
+    loadButton();
+  }, [sessionFromURL]);
+
+  useEffect(() => {
+    const updateInvolvements = async () => {
+      setLoadingInvolvements(true);
+      setAllInvolvements(await involvementService.getAll(selectedSession));
+
+      if (profile) {
+        const memberships = await membershipService.get({
+          username: profile.AD_Username,
+          sessionCode: selectedSession,
+          participationTypes: ['MEMBR', 'LEAD', 'ADV', 'GUEST'],
+        });
+        setMyInvolvements(memberships);
+      }
+
+      setLoadingInvolvements(false);
+    };
+
+    if (selectedSession) {
+      updateInvolvements();
+    }
+  }, [selectedSession, profile]);
+
+  const handleCropSubmit = (imageData) => {
+    setCroppedImage(imageData);
+  };
+
+  const clearOnClose = () => {
+    setOpenUploadForm(false);
+    setCroppedImage(null);
+  };
+
+  // NEW: Delete dialog logic
+  const handleDeleteClick = (poster) => {
+    setPosterToDelete(poster);
+    setOpenDeleteDialog(true);
+  };
+
+  const handleConfirmDelete = async () => {
+    if (posterToDelete) {
+      await hidePoster(posterToDelete.ID);
+      setAllPosters((prev) => prev.filter((p) => p.ID !== posterToDelete.ID));
+      setOpenDeleteDialog(false);
+      setPosterToDelete(null);
+      createSnackbar('Poster deleted successfully', 'success');
+    }
+  };
+
+  const handleCancelDelete = () => {
+    setOpenDeleteDialog(false);
+    setPosterToDelete(null);
+  };
+
+  const getClubName = (involvementCode) => {
+    const involvement = allInvolvements.find((inv) => inv.InvolvementCode === involvementCode);
+    return involvement ? involvement.Name : involvementCode;
+  };
+
+  // Removed debug logging to avoid console noise in production
```

Comment:
Seems like it would be good to delete this comment, since it belonged in a commit message or pull request.